service-runner.js

```diff
@@ -178,19 +204,35 @@ ServiceRunner.prototype._runMaster = function() {
     // Set up rolling restarts
     process.on('SIGHUP', this._rollingRestart.bind(this));
 
-    return this._startWorkers(this.config.num_workers);
+    return this._startWorkers(this.config.num_workers)
+    .then(function(workers) {
+        setTimeout(function() {
```

Comment:
Honestly, this feels like a nasty (and random) work-around. How about starting `self._checkHeartBeat()` as soon as the master is up, but then adding worker keys only once they've sent the `startup_finished` message?